pkg/k8s/apps/deployments.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/okteto/okteto/pkg/k8s/annotations"
+	"github.com/okteto/okteto/pkg/k8s/deployments"
+	"github.com/okteto/okteto/pkg/k8s/pods"
+	"github.com/okteto/okteto/pkg/k8s/replicasets"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/pointer"
+)
+
+type DeploymentApp struct {
```

Comment:
Some comments on these new implementations and interfaces may help. I also find it great to mention when something is made and we know we want it to implement some interface.

Reply:
It may allow you to remove some of the receivers or at least simplify them a tiny bit if embedding is used, https://golang.org/doc/effective_go#embedding

```suggestion
type DeploymentApp struct {
        *appsv1.Deployment
}
```

Then things like Name could be accessed directly, foo := NewDeploymentApp(d); fmt.Println(foo.Name)

Reply:
I tried embedding first, but had problems with methods like:
https://github.com/okteto/okteto/pull/1765/files#diff-b97445ee5f3a92ab765be2bc8402d8291b41a4acd087fa16e09d159c79c4a367R262

I will give it another thought for the next iteration